File: app/src/main/java/com/example/mvvmtest/view/fragment/UserViewDiscoverFragment.java
package com.example.mvvmtest.view.fragment;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.bumptech.glide.Glide;
import com.example.mvvmtest.R;
import com.example.mvvmtest.interfaces.OnUserViewDiscoverFragmentActionListener;
import com.example.mvvmtest.model.DiscoverUser;
import com.example.mvvmtest.util.DiscoverAction;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

public class UserViewDiscoverFragment extends Fragment {

    @BindView(R.id.image1DU)
    protected ImageView image1;
    @BindView(R.id.image2DU)
    protected ImageView image2;
    @BindView(R.id.image3DU)
    protected ImageView image3;
    @BindView(R.id.zodiacDU)
    protected ImageView zodiac;
    @BindView(R.id.nameDU)
    protected TextView name;
    @BindView(R.id.cityDU)
    protected TextView city;
    @BindView(R.id.homeTownDU)
    protected TextView hometown;
    @BindView(R.id.professionDU)
    protected TextView profession;
    @BindView(R.id.jobDU)
    protected TextView job;
    @BindView(R.id.heightDU)
    protected TextView height;
    @BindView(R.id.noideaDU)
    protected TextView noIdea;
    @BindView(R.id.drinkDU)
    protected TextView drink;
    @BindView(R.id.smokeDU)
    protected TextView smoking;
    @BindView(R.id.btnDislike)
    protected FloatingActionButton btnDislike;
    @BindView(R.id.btnlike)
    protected FloatingActionButton btnLike;
    @BindView(R.id.btnILoveIt)
    protected FloatingActionButton btnIloveIt;

    private DiscoverUser user;
    private OnUserViewDiscoverFragmentActionListener onActionListener;

    public static UserViewDiscoverFragment newInstance(DiscoverUser user) {
        UserViewDiscoverFragment fragment = new UserViewDiscoverFragment();
        Bundle args = new Bundle();
        args.putParcelable("user", user);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        user = getArguments().getParcelable("user");
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.user_view_discover_fragment, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        processUser();
    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        if (getParentFragment() instanceof OnUserViewDiscoverFragmentActionListener) {
            onActionListener = (OnUserViewDiscoverFragmentActionListener) getParentFragment();
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnUserViewDiscoverFragmentActionListener");
        }
    }

    private void processUser() {
        processImages();
        processText();
    }

    private void processImages() {
        List<String> images = user.getImages();
        if (images == null) {
            return;
        }
        //todo hacer logica para la cantidad de imagenes que bajaron
        //todo procesar zodiaco
        Glide.with(getContext())
                .asBitmap()
                .load(images.get(0))
                .into(image1);

        Glide.with(getContext())
                .asBitmap()
                .load(images.get(1))
                .into(image2);

        Glide.with(getContext())
                .asBitmap()
                .load(images.get(2))
                .into(image3);
    }

    private void processText() {
        name.setText(user.getCity() + " " + user.getAge());
        city.setText(user.getCity());
        hometown.setText(user.getHometown());
        profession.setText(user.getProfession());
        job.setText(user.getJob());
        height.setText(user.getHeight());
        drink.setText(user.getSmoke().name());
        smoking.setText(user.getSmoke().name());
    }

    @OnClick(R.id.btnILoveIt)
    void IloveItClicked(){
        onActionListener.onUserAction(user.getIdUser(), DiscoverAction.I_LOVE_IT);
    }


}

File: app/src/main/java/com/example/mvvmtest/ui/fragment/UserViewDiscoverFragment.kt
package com.example.mvvmtest.ui.fragment

import android.content.Context
import android.os.Bundle
import androidx.fragment.app.Fragment

import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.bumptech.glide.Glide
import com.example.mvvmtest.R
import com.example.mvvmtest.interfaces.OnUserViewDiscoverFragmentActionListener
import com.example.mvvmtest.model.DiscoverUser
import com.example.mvvmtest.util.DiscoverAction
import butterknife.OnClick
import kotlinx.android.synthetic.main.user_view_discover_fragment.*

class UserViewDiscoverFragment : Fragment(), View.OnClickListener {

    private var user: DiscoverUser? = null
    private var onActionListener: OnUserViewDiscoverFragmentActionListener? = null

    companion object {

        fun newInstance(user: DiscoverUser): UserViewDiscoverFragment {
            val fragment = UserViewDiscoverFragment()
            val args = Bundle()
            args.putParcelable("user", user)
            fragment.arguments = args
            return fragment
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        user = arguments!!.getParcelable("user")
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.user_view_discover_fragment, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initListeners()
        processUser()
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        if (parentFragment is OnUserViewDiscoverFragmentActionListener) {
            onActionListener = parentFragment as OnUserViewDiscoverFragmentActionListener?
        } else {
            throw RuntimeException("$context must implement OnUserViewDiscoverFragmentActionListener")
        }
    }

    private fun processUser() {
        processImages()
        processText()
    }

    private fun initListeners() {
        btnILoveIt.setOnClickListener(this)
        btnDislike.setOnClickListener(this)
        btnlike.setOnClickListener(this)
    }

    private fun processImages() {
        //todo hacer logica para la cantidad de imagenes que bajaron
        //todo procesar zodiaco
        Glide.with(activity!!)
                .asBitmap()
                .load(R.drawable.leo)
                .into(zodiacDU)

        Glide.with(context!!)
                .asBitmap()
                .load(R.drawable.image1)
                .into(image1DU)

        Glide.with(context!!)
                .asBitmap()
                .load(R.drawable.image2)
                .into(image2DU)

        Glide.with(context!!)
                .asBitmap()
                .load(R.drawable.image3)
                .into(image3DU)
    }

    private fun processText() {
        nameDU.text = "Aproximadamente a ${user!!.distance} km"
        cityDU.text = user!!.city
        homeTownDU.text = user!!.hometown
        professionDU.text = user!!.profession
        jobDU.text = user!!.job
        heightDU.text = user!!.height
        drinkDU.text = "Sometimes"
        smokeDU.text = "Sometimes"
    }

    override fun onClick(v: View?) {
        when (v?.id) {
            R.id.btnILoveIt ->
                onActionListener!!.onUserAction(user!!.idUser, DiscoverAction.I_LOVE_IT)
            R.id.btnDislike ->
                onActionListener!!.onUserAction(user!!.idUser, DiscoverAction.DISLIKE)
            R.id.btnlike ->
                onActionListener!!.onUserAction(user!!.idUser, DiscoverAction.LIKE)
            else -> {
            }
        }
    }

}
